File: Complementar - Estruturas de dados (Didático)/Algoritmos06/Alg06.rb
=begin
♦ Algoritmo 06: Algoritmo de Classificação/Ordenação para Estruturas de Dados Lineares Estáticas (não Dinâmicas): Selection Sort (Ordenação por Seleção).
=end

def selection_sort(array)
    n = array.length
  
    for i in 0..n-2
      indice_menor = i
  
      for j in i+1..n-1
        if array[j] < array[indice_menor]
          indice_menor = j
        end
      end
  
      array[indice_menor], array[i] = array[i], array[indice_menor]
    end
  
    return array
end
  
array = [64, 34, 25, 12, 22, 11, 90]

array_ordenado = selection_sort(array)

puts "Array ordenado:"
puts array_ordenado.join(" ")
  

File: Complementar - Estruturas de dados (Didático)/Algoritmos06/Alg06.py
'''
♦ Algoritmo 06: Estrutura de Dados Linear Estática (não Dinâmica) de Classificação/Ordenação Selection Sort (Ordenação pelo Método (Iterativo) de Seleção).
'''

def selectionSort(array):
    n = len(array)
    
    for i in range(n - 1):
        indiceMenor = i
        
        for j in range(i + 1, n):
            if array[j] < array[indiceMenor]:
                indiceMenor = j
        
        array[indiceMenor], array[i] = array[i], array[indiceMenor]

array = [64, 34, 25, 12, 22, 11, 90]

selectionSort(array)

print("Array ordenado:")
for element in array:
    print(element, end=" ")
